test/unit/test_table.c

```diff
@@ -106,15 +106,58 @@ bool test_r_table_sort1(void) {
 	mu_end;
 }
 
-bool test_r_table_columns() {
+bool test_r_table_uniq(void) {
+	RTable *t = __table_test_data1 ();
+
+	r_table_uniq (t);
+	char *strd = r_table_tostring (t);
+	mu_assert_streq (strd,
+		"ascii code\n"
+		"----------\n"
+		"a     97\n"
+		"b     98\n"
+		"c     99\n",
+		"uniq delete nothing");
+	free (strd);
+
+	r_table_add_row (t, "a", "97", NULL);
+	r_table_add_row (t, "a", "97", NULL);
+	r_table_add_row (t, "a", "97", NULL);
+	r_table_add_row (t, "b", "98", NULL);
+	r_table_add_row (t, "c", "99", NULL);
+	r_table_add_row (t, "b", "98", NULL);
+	r_table_add_row (t, "c", "99", NULL);
+	r_table_add_row (t, "d", "99", NULL);
+	r_table_add_row (t, "b", "98", NULL);
+	r_table_add_row (t, "d", "99", NULL);
+	r_table_add_row (t, "c", "99", NULL);
+	r_table_add_row (t, "c", "100", NULL);
+
+	r_table_uniq (t);
```

Comment:
imho this API should specify the column to be "uniq" instead of comparing them all. and there must be a r_table_query syntax for this. ideas here?

Reply:
mmm, this API seems like is the "DISTINCT" in sql, right? if so, I think it is alright to not have any column.

What you mean is  something like "GROUP BY" in  sql, I guess.